### PR TITLE
35665 Move Data Analysis GUI to Inelastic Tab Category

### LINKED ISSUE #35665: Move Data analysis GUI
In the indirect GUI's there is a "data analysis" GUI. However, it is also used for direct experiments. This issue is to move it to the inelastic section (including files and docs updated).

### PR DESCRIPTION
**Pull request by adriazalvarez**
### Description of work
Moved Analysis GUI: Moved the  'Data Analysis' interface category from Direct and Indirect into Inelastic. Added new release documentation in inelastic.
#### Summary of work

Part of #35665. 

### To test:

Opening mantid-workbench GUI / menu interfaces should show Data Analysis interface only on the Inelastic category menu. 

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Comment by jhaigh0** [comment]
This will cause one of my new tests to fail (haven't been merged yet). So one of us will have to edit the test depending on which gets merged first.

https://github.com/mantidproject/mantid/pull/36168

File: docs/source/release/v6.9.0/Inelastic/New_features/35665.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+- Moved 'Data Analysis' interface from Indirect and Direct interface menus into Inelastic interface menu.
```
Comment (robertapplin) [inline docs/source/release/v6.9.0/Inelastic/New_features/35665.rst:1]: It is possible to add a link to the Data Analysis documentation file by doing something like this:
```suggestion
- Moved :ref:`Data Analysis <interface-inelastic-data-analysis>` interface from Indirect and Direct interface menus into Inelastic interface menu.
```

**Review by robertapplin — CHANGES_REQUESTED** [review]
The changes look good and the Data Analysis interface appears in the `Inelastic` category as expected

The last thing to do on this PR is to move the documentation file [here](https://github.com/mantidproject/mantid/blob/main/docs/source/interfaces/indirect/Inelastic%20Data%20Analysis.rst) to a new `inelastic` folder in this location https://github.com/mantidproject/mantid/tree/main/docs/source/interfaces

While we do this, we might as well move the following two files to this location too as they both relate to inelastic interfaces:
https://github.com/mantidproject/mantid/blob/main/docs/source/interfaces/indirect/QENS%20Fitting.rst
https://github.com/mantidproject/mantid/blob/main/docs/source/interfaces/indirect/Inelastic%20Data%20Manipulation.rst

**Comment by robertapplin** [comment]
We will also need to link to the new `inelastic` folder location in this index file https://github.com/mantidproject/mantid/blob/main/docs/source/interfaces/index.rst

**Comment by jhaigh0** [comment]
My system test has merged in now, so you may need to edit this list
https://github.com/mantidproject/mantid/blob/main/Testing/SystemTests/tests/framework/InterfaceDocumentationTest.py#L23-L24

Essentially, that list is for documentation that we know doesn't exist. Normally it means that we need to make the documentation then remove the list entry, but in this case the test is looking for two docs that don't exist because those links both go to Inelastic Data Analysis instead of what they day (Direct Data Analysis, Indirect Data Analysis).

I'm unsure on the details of this PR but if there is still a path in the Interfaces menu that goes Direct -> Data Analysis, or Indirect -> Data Analysis, then those lines need to stay. But if they're removed and replaced with Inelastic -> Data Analysis then they can be deleted and providing there is a `Inelastic Data Analysis.rst` doc somewhere the test will pass.

**Review by robertapplin — APPROVED** [review]
The changes look good, and Data Analysis now appears in the Inelastic category.

All the tests have passed (crucially the `CppInterfacesStartupTest` system test has passed). It is hanging on the `PythonInterfacesStartupTest` for a reason unrelated to this PR, so please go ahead and merge this PR gatekeepers. I'll be fixing the `PythonInterfacesStartupTest` separately in #36331

**Pull request merged**